sys/taskqueue.c

```diff
@@ -48,8 +48,16 @@ void taskqueue_run(taskqueue_t *tq) {
   mtx_lock(&tq->tq_mutex);
   while (!STAILQ_EMPTY(&tq->tq_list)) {
     task = STAILQ_FIRST(&tq->tq_list);
+    mtx_unlock(&tq->tq_mutex);
+
     task->func(task->arg);
```

Comment:
Instead of repeatedly locking and unlocking the mutex it's better to copy head structure aside and reinitialize the original list head.